datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -65,23 +61,17 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
         for(String storeId: id)
         {
             book.setStoreID(Long.parseLong(storeId));
-            URL url;
             try{
-                url = new URL(map.get(Long.parseLong(storeId)) + "/books");
+                URL url = new URL(map.get(Long.parseLong(storeId)) + "/books");
```

Comment:
Im not clear about the point of this? If you want to check if the map has the storeId, why not just check that directly?